engines/crypto/aws/awskms.go

```diff
@@ -232,13 +237,177 @@ func (p *AWSKMSCryptoEngine) createPrivateKey(keySpec types.KeySpec) (string, cr
 }
 
 func (p *AWSKMSCryptoEngine) ImportRSAPrivateKey(key *rsa.PrivateKey) (string, crypto.Signer, error) {
-	lAWSKMS.Warnf("KMS does not support asymmetric key import. See https://docs.aws.amazon.com/kms/latest/developerguide/importing-keys.html")
-	return "", nil, fmt.Errorf("KMS does not support asymmetric key import")
+	var spec types.KeySpec
+	keySize := key.PublicKey.N.BitLen()
+	switch keySize {
+	case 2048:
+		spec = types.KeySpecRsa2048
+	case 3072:
+		spec = types.KeySpecRsa3072
+	case 4096:
+		spec = types.KeySpecRsa4096
+	default:
+		err := fmt.Errorf("key size not supported by AWS KMS")
+		lAWSKMS.Error(err)
+		return "", nil, err
+	}
+
+	createKeyOut, err := p.kmscli.CreateKey(context.Background(), &kms.CreateKeyInput{
```

Comment:
ImportRSAPrivateKey and ImportECDSAPrivateKey are almost the same. Is it posible to dedup the code?